src/js/screen_gfx.js

```diff
@@ -38,58 +38,79 @@ export function getPixelRatio() {
   return (typeof window !== "undefined" && window.devicePixelRatio) > 1 ? 2 : 1;
 }
 
-// Replaces `sourceVal` with `lightenVal` in inverse proportion to the brightness;
-// i.e. white remains white, black becomes `lightenVal`, and anit-aliased pixels remain anit-aliased
-export function lightenedColor(sourceVal, lightenVal) {
-  return 255 - (1 - sourceVal / 255) * (255 - lightenVal);
+// Replaces `sourceVal` with a blend of `lightenVal` and `darkenVal` proportional to the brightness;
+// i.e. white becomes `darkenVal`, black becomes `lightenVal`, and anit-aliased pixels remain anit-aliased
+export function blendedColorVal(sourceVal, lightenVal, darkenVal) {
+  return (
+    255 -
+    (sourceVal / 255) * (255 - darkenVal) -
+    (1 - sourceVal / 255) * (255 - lightenVal)
+  );
+}
+
+export function copyPixel(source, dest, sourceOffset, destOffset) {
+  //Red
+  dest[destOffset] = source[sourceOffset];
+  //Green
+  dest[destOffset + 1] = source[sourceOffset + 1];
+  //Blue
+  dest[destOffset + 2] = source[sourceOffset + 2];
+  //Alpha
+  dest[destOffset + 3] = source[sourceOffset + 3];
```

Comment:
Can we just use the spread operator here?

```javascript
let a = [1,2,3];
let b = [...a];
a[0] = 999;
console.log(a);
console.log(b);
```

Reply:
Ah ok maybe not the spread operator since it's from a specific position, but I assume there's a simpler way to array copy.

Reply:
Probably. I'm inclined to prioritize readability over cleverness unless there's a performance gain.